### PR TITLE
Provide more details to admins.

### PR DESCRIPTION
**Pull request by pmtischler**
For #54

### CODE REVIEW

File: server/auvsi_suas/models/target.py [hunk 1]
```diff
@@ -174,30 +174,10 @@ class Target(models.Model):
...
-                           name=self.__class__.__name__,
-                           thumbnail=self.thumbnail,
-                           thumbnail_approved=str(self.thumbnail_approved),
-                           creation_time=self.creation_time,
-                           last_modified_time=self.last_modified_time,
-                           **d))
-
-    def json(self):
+        d = self.json(is_admin=True)
+        return unicode(', '.join('%s=%s' % (k, v) for k, v in d.iteritems()))
```
Comment (prattmic) [inline server/auvsi_suas/models/target.py]: Don't you want a class name on the front? Or at least some kind of surrounding brackets?

File: server/auvsi_suas/models/target.py [hunk 2]
```diff
@@ -247,3 +227,19 @@ def json(self):
             'description': description,
             'autonomous': self.autonomous,
         }
+
+        if is_admin:
+            thumbnail = None
+            if self.thumbnail is not None:
+                thumbnail = self.thumbnail
+            d['thumbnail'] = thumbnail
```
Comment (prattmic) [inline server/auvsi_suas/models/target.py]: What am I missing? Why not `d['thumbnail'] = self.thumbnail`?

File: server/auvsi_suas/models/target_test.py [hunk 3]
```diff
@@ -135,13 +135,22 @@ def test_json(self):
...
         self.assertEqual(True, d['autonomous'])
+        self.assertNotIn('thumbnail_approved', d)
+
+        d = t.json(is_admin=True)
+        self.assertIn('thumbnail_approved', d)
+
+        t.thumbnail_approved = True
+        t.save()
+        d = t.json(is_admin=True)
+        self.assertEqual(True, d['thumbnail_approved'])
```
Comment (prattmic) [inline server/auvsi_suas/models/target_test.py]: Check for None thumbnail?

File: server/auvsi_suas/models/target.py [hunk 4]
```diff
@@ -174,30 +174,10 @@ class Target(models.Model):
...
-                           thumbnail_approved=str(self.thumbnail_approved),
-                           creation_time=self.creation_time,
-                           last_modified_time=self.last_modified_time,
-                           **d))
-
-    def json(self):
+        d = self.json(is_admin=True)
+        return unicode(', '.join('%s=%s' % (k, v) for k, v in d.iteritems()))
+
+    def json(self, is_admin):
```
Comment (prattmic) [inline server/auvsi_suas/models/target.py]: why not default is_admin to False?

File: server/auvsi_suas/views/targets.py [hunk 5]
```diff
@@ -172,7 +172,7 @@ def post(self, request):
                    autonomous=data.get('autonomous', False))
         t.save()
 
-        return JsonResponse(t.json(), status=201)
+        return JsonResponse(t.json(request.user.is_superuser), status=201)
```
Comment (prattmic) [inline server/auvsi_suas/views/targets.py]: this is inconsistent to the above, which uses the kwarg.

File: server/auvsi_suas/views/targets.py [hunk 6]
```diff
@@ -212,7 +212,7 @@ def get(self, request, pk):
         except ValueError as e:
             return HttpResponseForbidden(str(e))
 
-        return JsonResponse(target.json())
+        return JsonResponse(target.json(request.user.is_superuser))
 
     def put(self, request, pk):
```
Comment (prattmic) [inline server/auvsi_suas/views/targets.py]: Are we sure we want to allow superusers to modify other users' targets? Perhaps we should log which user last modified a target.

**Comment by prattmic** [comment]
A bunch of small comments, otherwise LGTM

File: server/auvsi_suas/models/target.py (hunk 1 above)
Comment (pmtischler) [inline server/auvsi_suas/models/target.py]: Ooops! Good catch.

File: server/auvsi_suas/models/target.py (hunk 4 above)
Comment (pmtischler) [inline server/auvsi_suas/models/target.py]: Done.

File: server/auvsi_suas/models/target.py (hunk 2 above)
Comment (pmtischler) [inline server/auvsi_suas/models/target.py]: Done.

File: server/auvsi_suas/models/target_test.py (hunk 3 above)
Comment (pmtischler) [inline server/auvsi_suas/models/target_test.py]: Done.

File: server/auvsi_suas/views/targets.py (hunk 5 above)
Comment (pmtischler) [inline server/auvsi_suas/views/targets.py]: Done.

File: server/auvsi_suas/views/targets.py (hunk 6 above)
Comment (pmtischler) [inline server/auvsi_suas/views/targets.py]: A superuser already can via the Django admin interface in an unlogged fashion.

On the other hand, I've decided to make a more narrow interfaces anyway for the judging interface, so I've reverted.

**Pull request merged**